src/main/java/com/opsgenie/plugin/graylog/OpsGenieGraylogClient.java

```diff
@@ -129,14 +131,25 @@ private CreateAlertRequest generateRequest(Stream stream, AlertCondition.CheckRe
         if (checkResult.getMatchingMessages().size() > 0) {
             stringBuilder.append("\nMatching messages: \n");
             for (MessageSummary summary : checkResult.getMatchingMessages()) {
-
                 stringBuilder.append("Message: ").append(summary.getMessage()).append("\n")
                         .append("Source: ").append(summary.getSource()).append("\n")
                         .append("Stream Ids: ").append(summary.getStreamIds()).append("\n")
-                        .append("Fields: ").append(summary.getFields()).append("\n\n")
-                        .append("------------------").append("\n")
+                        .append("Fields: \n")
                 ;
-
+                if(showFields != null && !showFields.trim().isEmpty()){
+                    summary.getFields().keySet()
+                        .iterator()
+                        .forEachRemaining(
+                            fieldKey -> {
+                                if(Arrays.asList(showFields.split(",")).contains(fieldKey)){
```

Comment:
Minor : you dont need to create `Arrays.asList(showFields.split(","))` everytime in the loop, can just create once before the loop